components/arm/client.go

```diff
@@ -39,21 +40,17 @@ func NewClientFromConn(
 	logger golog.Logger,
 ) (Arm, error) {
 	pbClient := pb.NewArmServiceClient(conn)
-	// TODO(DATA-853): requires that this support models being changed on the fly, not just at creation
-	// TODO(RSDK-882): will update this so that this is not necessary
-	r := robotpb.NewRobotServiceClient(conn)
-	model, modelErr := getModel(ctx, r, name.ShortName())
-	if modelErr != nil {
-		logger.Errorw("error getting model for arm; will not allow certain methods")
-	}
 	c := &client{
 		Named:  name.PrependRemote(remoteName).AsNamed(),
 		name:   name.ShortName(),
 		client: pbClient,
 		logger: logger,
 	}
-	if modelErr == nil {
-		c.model = model
+	clientFrame, err := c.updateKinematics(ctx)
+	if err != nil {
+		logger.Errorw("error getting model for arm; will not allow certain methods")
+	} else {
+		c.model = clientFrame
```

Comment:
Nothing special needs to be done here. Since Geometries is also implemented, that is called in the case where geometries need to be provided.

If extraneous geometries are added to an arm, I think the receiving driver would need to be responsible for placing them into the kinematic model if desired as well, since the individual driver would know how it wants to do so. The arm client shouldn't be making the decision on how to resolve that.